src/schema/tablecompiler.js

```diff
@@ -66,7 +66,7 @@ TableCompiler.prototype.alter = function () {
 
 TableCompiler.prototype.foreign = function (foreignData) {
   if (foreignData.inTable && foreignData.references) {
-    var keyName = this._indexCommand('foreign', this.tableNameRaw, foreignData.column);
+    var keyName = foreignData.keyName || this._indexCommand('foreign', this.tableNameRaw, foreignData.column);
```

Comment:
Should I wrap foreignData.keyName with `this.formatter.wrap()`?
If you think this feature is likely to be accepted, I'll be happy to update the PR.